eventuals/do-all.h

```diff
@@ -47,7 +50,21 @@ struct _DoAll {
                   }
                   if (counter_.fetch_sub(1) == 1) {
                     // You're the last eventual so call the continuation.
-                    k_.Start(std::move(values_));
+                    std::optional<std::exception_ptr> exception = GetExceptionIfExists();
+
+                    if (exception) {
+                      try {
+                        std::rethrow_exception(*exception);
+                      } catch (const StoppedException&) {
+                        k_.Stop();
+                      } catch (...) {
+                        k_.Fail(std::current_exception());
+                      }
+                    } else {
+                      k_.Start(GetTupleOfValues());
+                    }
+                  } else {
+                    interrupt_.Trigger();
                   }
```

Comment:
We don't want to interrupt if an eventual was successful, just want to wait for the rest!
```suggestion
                  }
```